plasmapy/diagnostics/charged_particle_radiography/synthetic_radiography.py

```diff
@@ -755,53 +677,14 @@ def load_particles(
     # Run/push loop methods
     # *************************************************************************
 
-    def _adaptive_dt(self, Ex, Ey, Ez, Bx, By, Bz):  # noqa: ARG002
-        r"""
-        Calculate the appropriate dt for each grid based on a number of
-        considerations
-        including the local grid resolution (ds) and the gyroperiod of the
-        particles in the current fields.
+    def _update_x_and_v_all(self):
         """
-        # If dt was explicitly set, skip the rest of this function
-        if self.dt.size == 1:
-            return self.dt
-
-        # candidate timesteps includes one per grid (based on the grid resolution)
-        # plus additional candidates based on the field at each particle
-        candidates = np.ones([self.nparticles_grid, self.num_grids + 1]) * np.inf
-
-        # Compute the timestep indicated by the grid resolution
-        ds = np.array([grid.grid_resolution.to(u.m).value for grid in self.grids])
-        gridstep = 0.5 * (ds / self.vmax)
-
-        # Wherever a particle is on a grid, include that grid's gridstep
-        # in the list of candidate timesteps
-        for i, _grid in enumerate(self.grids):  # noqa: B007
-            candidates[:, i] = np.where(self.on_grid[:, i] > 0, gridstep[i], np.inf)
-
-        # If not, compute a number of possible timesteps
-        # Compute the cyclotron gyroperiod
-        Bmag = np.max(np.sqrt(Bx**2 + By**2 + Bz**2)).to(u.T).value
-        # Compute the gyroperiod
-        if Bmag == 0:
-            gyroperiod = np.inf
-        else:
-            gyroperiod = 2 * np.pi * self.m / (self.q * np.max(Bmag))
-
-        candidates[:, self.num_grids] = gyroperiod / 12
-
-        # TODO: introduce a minimum timestep based on electric fields too!
-
-        # Enforce limits on dt
-        candidates = np.clip(candidates, self.dt[0], self.dt[1])
-
-        # dt is the min of all the candidates for each particle
-        # a separate dt is returned for each particle
-        dt = np.min(candidates, axis=-1)
-
-        # dt should never actually be infinite, so replace any infinities
-        # with the largest gridstep
-        return np.where(dt == np.inf, np.max(gridstep), dt)
+        Update the x_all and v_all quantities.
+        This function is called after each push to ensure the position and velocities
+        properties contain up-to-date quantities.
+        """
+        self.x_all = np.concatenate((self.x_not_tracked, self.Tracker.x))
+        self.v_all = np.concatenate((self.v_not_tracked, self.Tracker.v))
 
     def _coast_to_grid(self):
```

Comment:
Problem to consider if you want: is there a better way of evolving particles to the grid so they all go right to the grid, rather than just the 'first' particle?